adoc/deployment-azure.adoc

```diff
@@ -0,0 +1,232 @@
+using the ssh key specified via the `admin_ssh_key` variable.
+It's also possible to enable password based authentication by specifying a
+value for the `admin_password` variable. Note well: Azure has some security
+checks in place to avoid the usage of weak passwords.
+
+When the bastion host creation is disabled the access to the master nodes of
+the cluster is just a matter of doing a ssh against their public IP address.
+
+Accessing a cluster through an Azure Bastion requires a different procedure.
+
+==== Using Azure Bastion
+
+Azure bastion supports only RSA keys with PEM format. These can be created by
+doing:
+
+```
+ssh-keygen -f azure -t rsa -b 4096 -m pem
+```
+
+This will create a public and private key named `azure`. The **private** key has
+to be provided later to Azure, hence it's strongly recommended to create
+a dedicated key pair.
+
+Once the whole infrastructure is created you can connect into any node of the
+cluster by doing the following steps:
+
+  * Log into Azure portal
+  * Choose one of the nodes of the cluster
+  * Click "connect" and select "bastion" as option
+  * Enter all the required fields
+
+Once this is done a new browser tab will be open with a shell session running
+inside of the desired node.
+
+It's recommended to use Chrome or Chromium during this process.
+
+You can ssh into the first master node of the cluster of drive the creation of
+a SUSE cluster from there using this guide link:https://docs.microsoft.com/en-us/azure/bastion/bastion-connect-vm-ssh[Connect using SSH to a Linux virtual machine using Azure Bastion]. Once this is done you can download a kubeconfig
+file and operate the cluster without having to go through the bastion host.
+
+Caveats of Azure Bastion:
+
+  * As of June 2020, the link:https://docs.microsoft.com/en-us/azure/bastion/bastion-overview#regions[Azure Bastion service] is not available in all Azure regions.
+  * By design it's not possible to leverage the bastion host without using the
+    ssh session embedded into the browser. This makes impossible to use tools like
+    `sftp` or `scp`.
+  * You have to rely on copy and paste to share data (like the `admin.conf` file
+    generated by skuba) between the remote nodes and your local system.
+    You can "rely" on `cat`, `base64` and a lot of copy and paste...
+  * `skuba` requires a private ssh key to connect to all the nodes of the cluster.
+    You have to upload the private key you specified at cluster creation
+    or create a new one inside of the first master node and copy that
+    around the cluster.
+
+=== Virtual Network Peering Support
+
+It is possible to join existing networks to the cluster.  It can be setup by adding a list of network IDs to `peer_virutal_network_id`.
+
+=== Enable Multiple Zones
+
+It is possible to enable multiple zones.  It can be set `enable_zone` to `true` and master/worker nodes will bd distributed sequentially based on zones defined in `azure_availability_zones`. 
```

Comment:
```suggestion
It is possible to enable multiple zones.  It can be set `enable_zone` to `true` and master/worker nodes will be distributed sequentially based on zones defined in `azure_availability_zones`. 
```